modulemd/v2/include/modulemd-2.0/modulemd-module-stream.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * This file is part of libmodulemd
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Fedora-License-Identifier: MIT
+ * SPDX-2.0-License-Identifier: MIT
+ * SPDX-3.0-License-Identifier: MIT
+ *
+ * This program is free software.
+ * For more information on the license, see COPYING.
+ * For more information on free software, see <https://www.gnu.org/philosophy/free-sw.en.html>.
+ */
+
+#pragma once
+
+#include <glib.h>
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: modulemd-module-stream
+ * @title: Modulemd.ModuleStream
+ * @stability: stable
+ * @short_description: A parent class for all versions of #ModulemdModuleStream
+ * objects.
+ */
+
+/**
+ * ModulemdModuleStreamVersionEnum:
+ * @MD_MODULESTREAM_VERSION_ONE: Represents v1 of the #Modulemd.ModuleStream
+ * metadata format.
+ * @MD_MODULESTREAM_VERSION_TWO: Represents v2 of the #Modulemd.ModuleStream
+ * metadata format.
+ * @MD_MODULESTREAM_VERSION_LATEST: Represents the highest-supported version of
+ * the #Modulemd.ModuleStream metadata format.
+ *
+ * Since: 2.0
+ */
+typedef enum
+{
+  MD_MODULESTREAM_VERSION_UNSET = 0,
+
+  MD_MODULESTREAM_VERSION_ONE = 1,
+  MD_MODULESTREAM_VERSION_TWO = 2,
+
+  MD_MODULESTREAM_VERSION_LATEST = MD_MODULESTREAM_VERSION_TWO
+} ModulemdModuleStreamVersionEnum;
+
+
+#define MODULEMD_TYPE_DEFAULTS (modulemd_defaults_get_type ())
+
+
+#define MODULEMD_TYPE_MODULE_STREAM (modulemd_module_stream_get_type ())
+
+G_DECLARE_DERIVABLE_TYPE (ModulemdModuleStream,
+                          modulemd_module_stream,
+                          MODULEMD,
+                          MODULE_STREAM,
+                          GObject)
+
+struct _ModulemdModuleStreamClass
+{
+  GObjectClass parent_class;
+
+  ModulemdModuleStream *(*copy) (ModulemdModuleStream *self,
+                                 const gchar *copy);
+
+  gboolean (*validate) (ModulemdModuleStream *self, GError **error);
+
+  guint64 (*get_mdversion) (ModulemdModuleStream *self);
+
+  /* Padding to allow adding up to 10 new virtual functions without
+   * breaking ABI. */
+  gpointer padding[10];
+};
+
+
+/**
+ * modulemd_module_stream_new:
+ * @mdversion: (in): The metadata version of ModuleStream to create.
+ * @module_name: (in): The name of the module.
+ * @stream_name: (in) (nullable): The name of this stream. Optional.
```

Comment:
Note that the spec calls this `module_stream`, and that's the name we used in other objects too.
Not a big deal, but just consistency note.

Reply:
Agreed, we should be consistent. Fixed.